### PR TITLE
[test] Modernize GREASY Tests

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

File: cscs-checks/apps/greasy/greasy_check.py [hunk 1]
```diff
@@ -16,79 +16,70 @@ def to_seconds(str):
...
+                               ('mpi', 'gpu', 24,  4, 3, 1),
+                               ('mpi', 'mc',  72, 12, 3, 1),
+                               ('mpi+openmp', 'gpu', 24,  3, 2, 2),
+                               ('mpi+openmp', 'mc',  72,  6, 3, 2)])
+    valid_prog_environs = ['PrgEnv-gnu']
+    sourcepath = 'tasks_mpi_openmp.c'
+    build_system = 'SingleSource'
+    executable = 'tasks_mpi_openmp.x'
+    tasks_file = 'tasks.txt'
+    greasy_logfile = 'greasy.log'
```
Comment (jjotero) [inline cscs-checks/apps/greasy/greasy_check.py:34]:
```suggestion
    tasks_file = variable(str, value='tasks.txt')
    greasy_logfile = variable(str, 'greasy.log')
```

File: cscs-checks/apps/greasy/greasy_check.py [hunk 2]
```diff
@@ -16,79 +16,70 @@ def to_seconds(str):
...
+    nnodes = 2
```
Comment (jjotero) [inline cscs-checks/apps/greasy/greasy_check.py:35]:
```suggestion
    nnodes = variable(int, value=2)
```

File: cscs-checks/apps/greasy/greasy_check.py [hunk 3]
```diff
@@ -16,79 +16,70 @@ def to_seconds(str):
...
+
+    # sleep enough time to distinguish if the files are running in parallel
+    # or not
+    sleep_time = 60
```
Comment (jjotero) [inline cscs-checks/apps/greasy/greasy_check.py:39]:
```suggestion
    sleep_time = variable(int, value=60)
```

File: cscs-checks/apps/greasy/greasy_check.py [hunk 4]
```diff
@@ -16,79 +16,70 @@ def to_seconds(str):
...
+    use_multithreading = False
+    modules = ['GREASY']
+    maintainers = ['VH', 'SK']
+    tags = {'production'}
+
+    @run_after('init')
+    def unpack_configuration_parameter(self):
+        self.variant, self.partition = self.configuration[0:2]
+        self.num_greasy_tasks, self.workers_per_node = self.configuration[2:4]
+        self.ranks_per_worker, self.cpus_per_worker = self.configuration[4:6]
```
Comment (jjotero) [inline cscs-checks/apps/greasy/greasy_check.py:49]: I'd probably also create typed variables for all these so you get type-checking on the parameters.

File: cscs-checks/apps/greasy/greasy_check.py [hunk 5]
```diff
@@ -16,79 +16,70 @@ def to_seconds(str):
...
-                'time': (refperf, None, 0.5, 's')
-            }
-        }
-        self.perf_patterns = {
-            'time': sn.extractsingle(r'Total time: (?P<perf>\S+)',
-                                     self.greasy_logfile,
-                                     'perf', to_seconds)
-        }
+    @run_before('run')
+    def set_variables(self):
```
Comment (jjotero) [inline cscs-checks/apps/greasy/greasy_check.py:75]:
```suggestion
    def set_environment_variables(self):
```

File: cscs-checks/apps/greasy/greasy_check.py [hunk 6]
```diff
@@ -114,7 +105,7 @@ def daint_dom_gpu_specific_workaround(self):
                 }
             }
         elif self.current_partition.fullname in ['daint:mc']:
-            if self.variant != 'serial':
+            if self.configuration.variant != 'serial':
```
Comment (jjotero) [inline cscs-checks/apps/greasy/greasy_check.py:108]:
```suggestion
            if 'serial' not in self.variant:
```

**Review by jjotero — CHANGES_REQUESTED** [review]

**Review by jjotero — APPROVED** [review]
lgtm

File: cscs-checks/apps/greasy/greasy_check.py [hunk 7]
```diff
@@ -234,3 +233,24 @@ def num_ranks(match):
         ))
 
         return True
+
+    @run_before('performance')
+    def set_perf_patterns(self):
```
Comment (vkarak) [inline cscs-checks/apps/greasy/greasy_check.py:238]: What about using the new `@performance_function` decorator that was just merged?

**Review by vkarak — APPROVED** [review]

**Pull request merged**